scripts/quests/windurst/Toraimarai_Turmoil.lua

```diff
@@ -0,0 +1,142 @@
+                        npcUtil.giveKeyItem(player, xi.ki.RHINOSTERY_CERTIFICATE)
+                    end
+                end,
+            },
+        },
+    },
+
+    {
+        --initial completion
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_ACCEPTED
+        end,
+
+        [xi.zone.WINDURST_WATERS] =
+        {
+            ['Ohbiru-Dohbiru'] =
+            {
+                onTrigger = function(player, npc)
+                    return quest:event(786, 4500, xi.keyItem.RHINOSTERY_CERTIFICATE, xi.item.STARMITE_SHELL) -- Reminder text.
+                end,
+
+                onTrade = function(player, npc, trade)
+                    if npcUtil.tradeHasExactly(trade, { { xi.item.STARMITE_SHELL, 3 } }) then
+                        return quest:progressEvent(791)
+                    end
+                end,
+            },
+
+            ['Leepe-Hoppe'] =
+            {
+                onTrigger = function(player, npc)
+                    return quest:event(790, 0, xi.ki.RHINOSTERY_CERTIFICATE)
+                end,
+            },
+
+            onEventFinish =
+            {
+                [791] = function(player, csid, option, npc)
+                    if quest:complete(player) then
+                        player:confirmTrade()
+                    end
+                end,
+            },
+        },
+
+        [xi.zone.WINDURST_WALLS] =
+        {
+            ['Polikal-Ramikal'] =
+            {
+                onTrigger = function(player, npc)
+                    return quest:event(391)
+                end,
+            },
+
+            ['Yoran-Oran'] =
+            {
+                onTrigger = function(player, npc)
+                    return quest:event(392)
+                end,
+            },
+        },
+    },
+
+    {
+        --repeat completion
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_COMPLETED
+        end,
+
+        [xi.zone.WINDURST_WATERS] =
+        {
+            ['Ohbiru-Dohbiru'] =
+            {
+                onTrigger = function(player, npc)
+                    return quest:event(795, 4500, 0, xi.item.STARMITE_SHELL) -- repeat dialog
+                end,
+
+                onTrade = function(player, npc, trade)
+                    if npcUtil.tradeHasExactly(trade, { { xi.item.STARMITE_SHELL, 3 } }) then
+                        return quest:progressEvent(791)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [791] = function(player, csid, option, npc)
+                    player:confirmTrade()
+
+                    local gilReward = 4500
+                    --From previous implementation, award 100 fame on first completion,
+                    -- and 50 fame for any subsequent trade.
+                    player:addFame(xi.fameArea.WINDURST, 50)
+                    player:addGil(gilReward)
```

Comment:
Messaging/adding gil can be simplified by using npcUtil.giveCurrency()

Reply:
Perfect, thanks!